pyccl/core.py

```diff
@@ -136,31 +151,47 @@ def __init__(self, Omega_c=None, Omega_b=None, h=None, A_s=None, n_s=None,
         if norm_pk < 1e-5 and sigma8 is not None:
             raise ValueError("sigma8 must be greater than 1e-5.")
         
+        if isinstance(m_nu, float):
+            if mnu_type == None :
+                mnu_type = 'sum'
+            m_nu = [m_nu]
+        elif hasattr(m_nu, "__len__"):
+            if (len(m_nu)!=3):
+                raise ValueError("m_nu must be a float or array-like object with length 3.")
+            elif ((mnu_type=='sum') or (mnu_type=='sum_inverted') or (mnu_type=='sum_equal')):
+                raise ValueError("That mnu type cannot be passed with a list of neutrino masses, only with a sum.");
```

Comment:
Replace `"That mnu type..."` with: `"mnu_type '%s' cannot be passed..." % mnu_type`, so it's clear to the user what setting has been detected.

Reply:
Fixed.